### PR TITLE
add glossary to documentation

### PR DESCRIPTION
**Pull request by dhellmann**
**What this PR does / why we need it**:

Move the glossary section out of the scope document into its own document
and format it so it is easier to link directly to specific terms from
other locations. This reduces the need to duplicate definitions in
multiple design documents and makes it more obvious where to write down
terms when we find communication issues.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

NONE

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

No images are changed.

**Release note**:

NONE

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
+---
+title: Cluster API Scope and Objectives
```
Comment (ncdc) [inline docs/glossary.md]: fixme :)

File: docs/glossary.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+authors:
+  - "@dhellmann"
+reviewers:
+  - "@timothysc"
```
Comment (detiber) [inline docs/glossary.md]: This should probably be the union of both cluster-api-admins and cluster-api-maintainers here: https://github.com/kubernetes-sigs/cluster-api/blob/master/OWNERS_ALIASES

Comment (dhellmann) [reply docs/glossary.md]: Should I update the authors, reviewers, or both?

File: docs/glossary.md (hunk 1 above)
Comment (dhellmann) [reply docs/glossary.md]: Done

File: docs/glossary.md (hunk 2 above)
Comment (timothysc) [reply docs/glossary.md]: I think it's fine for the creator of the doc, it's not an OWNERS file.

Comment (detiber) [reply docs/glossary.md]: Sorry, just the reviewers :)

File: docs/glossary.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+### Default implementation
+
+A feature implementation offered as part of the Cluster API project, infrastructure providers can swap it out for a different one.
+
+## H
+
+### Horizontal Scaling
+
+TBD
```
Comment (timothysc) [inline docs/glossary.md]: The ability to add more machines based on policy and well defined metrics.  For example, add a machine to a cluster when CPU load average > (X) for a period of time (Y).

File: docs/glossary.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
+---
+title: Glossary
+authors:
+  - "@dhellmann"
```
Comment (timothysc) [inline docs/glossary.md]: `- "@timothysc"`

File: docs/glossary.md (hunk 3 above)
Comment (timothysc) [reply docs/glossary.md]: should have added this earlier.

File: docs/glossary.md (hunk 2 above)
Comment (dhellmann) [reply docs/glossary.md]: Done

File: docs/glossary.md (hunk 4 above)
Comment (dhellmann) [reply docs/glossary.md]: Done

File: docs/glossary.md (hunk 3 above)
Comment (dhellmann) [reply docs/glossary.md]: How about if we leave adding that to another PR? I'd like everyone to get in the habit of updating this document as we go along and write other docs.

Comment (davidewatson) [reply docs/glossary.md]: The trick is to determine who will submit the PR and when? For non-trivial changes, I would recommend opening an issue. For trivial changes, it seems easier just to make the change and save the time of an additional lgtm/approve cycle.

**Comment by davidewatson** [comment]
/hold
/lgtm

The hold is for the remaining feedback from Tim.

File: docs/glossary.md (hunk 3 above)
Comment (davidewatson) [reply docs/glossary.md]: FWIW, I'd be willing to make this change today. Along with adding a link to the glossary to the GitBook...

Comment (dhellmann) [reply docs/glossary.md]: I'm still working out the process myself, so if you want me to make the changes here, that's fine. But I also don't mind filing a separate PR, or having you do it.

**Pull request merged**